### ISSUE TITLE
Panes

### ISSUE THREAD

**Issue by rjdestigter** [PaulLeCam/react-leaflet#219]
I'm looking at implementing Panes (`http://leafletjs.com/reference-1.0.0.html#map-pane`) in `react-leaflet` Panes were added to Leaflet v1 to allow more control of what is rendered on top of what. (zIndex)

Panes in Leaflet are just DOM nodes at the moment and not an instance of some form of class. A React component that creates and removes on mount and unmount should be fairly straight forward.

Passing `pane` as a prop to react-leaflet's Marker, GeoJson, Layer components will already work. Do you think I should implement `pane` as a context type just like`map` is?

**Comment by rjdestigter** [comment]
I've got an early working version available here: `https://github.com/rjdestigter/react-leaflet/tree/MapPane`

I added `pane` as a `contextType` to LayerGroup which in turn is passed on to `L.layerGroup`

Example:
```
<Map>
  <MapPane name='foodtrucks'>
    <LayerGroup>
       <MyFoodTruckMarkers />
   </LayerGroup>
  </MapPane>
</Map>
```

**Comment by PaulLeCam** [comment]
Hi,
Seems like a good idea, I won't have time to work on this for the next few weeks and I don't think it needs to be released in v1.0 but seems like a good addition to a minor release.

A few things I can think of:
- This new `Pane` component should only support the `className`, `name` and `style` props, no event handling or other hacks people could try to use it for, it would become very hard to maintain an expected behavior.
- The `name` prop should be optional in React-Leaflet and can be auto-generated, ex using `uniqueId()` from Lodash as it's done in the [Map component](https://github.com/PaulLeCam/react-leaflet/blob/master/src/Map.js#L60).
- Does Leaflet allow to override its existing panes?
  - If so, we could have ex `<Pane name='overlayPane' style={{zIndex: 620}} />` effectively changing the z-index of all overlays dynamically, assuming Leaflet applies the updates.
  - One thing to be very careful with is about removing such a Pane, would Leaflet reset its own?
  - If not supported or no way to properly remove the Pane, the existing Leaflet names should be blacklisted (= the component has no effect and displays a warning).
- Regarding the implementation, it's likely very similar to the Map component:
  - The Pane goes through a first render where it needs to render the `<div>` with a reference and prevent the children from rendering (as the `pane` wouldn't be set in context yet), as done in the [Map component](https://github.com/PaulLeCam/react-leaflet/blob/master/src/Map.js#L113).
  - In its `componentDidMount()` handler, the Pane needs to call `map.createPane()` and set the created `pane` in its state, to trigger a re-render. Again, same logic as the [Map component](https://github.com/PaulLeCam/react-leaflet/blob/master/src/Map.js#L64).
  - Subsequent renders should display the children components, now having access to the `pane` in context exposed by the Pane.

**Comment by PaulLeCam** [comment]
Closing as now in PR https://github.com/PaulLeCam/react-leaflet/pull/227
